### DISCUSSION TITLE
Sensitivity + Optimization + Surrogate model

### DISCUSSION THREAD

**Discussion by Val4on** [snl-dakota/dakota#21] [Q&A]
Hi Dakota community,

I'm looking for an example that does sensitivity analysis and then runs optimization with a surrogate model. 
I couldn't not find an example in the shipped examples directory, can someone explain how to do this or send me a sample Dakota setup? 

Thanks!

**Comment by jadamstephens** [comment]
Dakota does not support performing sensitivity analysis and optimization on a surrogate in a single study. However, it's fairly straightforward to do this in two studies.

One possible workflow:

1. Dakota builds a surrogate using data it collects from a simulation and performs sensitivity analysis, as in [this example](https://github.com/snl-dakota/dakota-examples/tree/master/official/surrogates/dace). For this step, ensure a `seed` is set for the DACE method (`DesignMethod` in the example input file.)
2. Note that a restart file named (by default) dakota.rst was created by Dakota in step 1. Hang on to that!
3. Modify the input file from step 1, changing the sensitivity analysis (`EvalSurrogate` block in the example) method to an optimization method. You'll also need to run Dakota with the `-read_restart` option and provide it the name of the restart file from step 1. When Dakota tries to run the DACE method to collect training data to build the surrogate, evaluation data from the restart file will be used and the simulation won't be run. For this to work correctly, it is critical that the DACE method have the same seed in this step that it did in step 1.
